### PR TITLE
62: implement IncrementsRelativeDLTCumulative

### LINKED ISSUE #62: An extension suggestion: IncrementsRelativeDLT.Cumulative.R
This is an extension suggestion. A new class which is a slight modification of the existing IncrementsRelativeDLT class to take into account all DLTs so far (cumulative approach) instead of DLTs observed at the current dose. See the attached txt file for the R-code and documentation.

[IncrementsRelativeDLTCumulative.txt](https://github.com/Roche/crmPack/files/7735941/IncrementsRelativeDLTCumulative.txt)

### PR DESCRIPTION
**Pull request by danielinteractive**
Closes #62

### CODE REVIEW

**Comment by danielinteractive** [comment]
@gunhanb you can just continue working on this branch and push updates. just ping me once you think it is ready for another chat

File: R/Rules-methods.R [hunk 1]
```diff
@@ -3340,4 +3327,39 @@ setMethod("windowLength",
...
+  def =
+    function(increments, data, ...) {
+
+      # Determine what was the last dose.
+      lastDose <- tail(data@x, 1)
+
+      # Determine how many DLTs have occurred in last cohort.
+      lastCohort <- tail(data@cohort, 1)
+      index <- which(data@cohort == lastCohort)
+      dltHappened <- sum(data@y[index])
```
Comment (danielinteractive) [inline R/Rules-methods.R:3353]: @gunhanb Question: would we not want instead to sum the DLTs across all cohorts, i.e. cumulative?

Comment (gunhanb) [reply R/Rules-methods.R:3353]: HI @danielinteractive you are abolsutely right. It is my mistake. The problem is about the documentation. What is implemented in crmPack is cumulative approach, but the extension is "not cumulative" actually.

Comment (gunhanb) [reply R/Rules-methods.R]: @danielinteractive Do you think now it can be merged?

**Review by gunhanb — APPROVED** [review]
Variables updated to be align with snake_cases

File: examples/Rules-class-IncrementsRelativeDLT.R [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
-# As example, here is the rule for: 
-#      maximum doubling the dose if no DLTs were observed at the current dose
-#      or maximum increasing the dose by 1.33 if 1 or 2 DLTs were observed at the current dose
-#      or maximum increasing the dose by 1.22 if 3 or more DLTs were observed
+# As example, here is the rule for:
+#      maximum doubling the dose if no DLTs were observed in the whole study so far
+#      or maximum increasing the dose by 1.33 if 1 or 2 DLTs were observed so far
+#      or maximum increasing the dose by 1.22 if 3 or more DLTs were observed so far
 
 myIncrements <- IncrementsRelativeDLT(DLTintervals = c(0, 1, 3),
```
Comment (danielinteractive) [inline examples/Rules-class-IncrementsRelativeDLT.R:7]: just need to fix these few linter warnings here

File: examples/Rules-class-IncrementsRelativeDLTCurrent.R [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+# As example, here is the rule for:
+#      maximum doubling the dose if no DLTs were observed at the current dose
+#      or maximum increasing the dose by 1.33 if 1 or 2 DLTs were observed at the current dose
+#      or maximum increasing the dose by 1.22 if 3 or more DLTs were observed
+
+my_Increments <- IncrementsRelativeDLTCurrent(DLTintervals = c(0, 1, 3),
```
Comment (danielinteractive) [inline examples/Rules-class-IncrementsRelativeDLTCurrent.R:7]:
```suggestion
my_increments <- IncrementsRelativeDLTCurrent(DLTintervals = c(0, 1, 3),
```
note: my_Increments would not be snake_case (lower case)

File: examples/Rules-method-maxDose-IncrementsRelativeDLTCurrent.R [hunk 4]
```diff
@@ -0,0 +1,21 @@
+
+# Create the data
+data <- Data(x=c(0.1, 0.5, 1.5, 3, 6, 10, 10, 10),
```
Comment (danielinteractive) [inline examples/Rules-method-maxDose-IncrementsRelativeDLTCurrent.R:3]: one easy way to fix these linter formatting errors often is to install the `styler` R-package and then in R studio to use the corresponding Addin to style a file. please have a try

**Review by danielinteractive — COMMENTED** [review]
thx Burak, pls see few comments inside to help with final cleanup of linter errors

**Comment by danielinteractive** [comment]
@gunhanb I am having a look now.

**Comment by danielinteractive** [comment]
@gunhanb it should work now, I fixed it

**Pull request merged**